res/controllers/Gemini-G4V-scripts.js

```diff
@@ -0,0 +1,562 @@
+// eslint-disable-next-line no-var
+var g4v = {};
+
+g4v.init = function() {
+    // Controls not linked to any mixer or deck
+    g4v.otherControls = new g4v.OtherControls();
+    // Four Mixer Strips
+    g4v.deck1Mixer = new g4v.MixerStrip(1);
+    g4v.deck2Mixer = new g4v.MixerStrip(2);
+    g4v.deck3Mixer = new g4v.MixerStrip(3);
+    g4v.deck4Mixer = new g4v.MixerStrip(4);
+    // Two decks
+    g4v.leftDeck = new g4v.Deck([1, 3], 1);
+    g4v.rightDeck = new g4v.Deck([2, 4], 2);
+
+    // TO ensure that efects are ready
+    engine.setParameter("[EffectRack1_EffectUnit1]", "group_[Channel1]_enable", 0);
+    engine.setParameter("[EffectRack1_EffectUnit2]", "group_[Channel2]_enable", 0);
+    engine.setParameter("[EffectRack1_EffectUnit3]", "group_[Channel3]_enable", 0);
+    engine.setParameter("[EffectRack1_EffectUnit4]", "group_[Channel4]_enable", 0);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "enabled", 1);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "enabled", 1);
+    engine.setParameter("[EffectRack1_EffectUnit3_Effect1]", "enabled", 1);
+    engine.setParameter("[EffectRack1_EffectUnit4_Effect1]", "enabled", 1);
+
+};
+
+g4v.shutdown = function()  {
+    // Shutdown components
+    g4v.otherControls.shutdown();
+    g4v.deck1Mixer.shutdown();
+    g4v.deck1Mixer.shutdown();
+    g4v.deck2Mixer.shutdown();
+    g4v.deck3Mixer.shutdown();
+    g4v.deck4Mixer.shutdown();
+    g4v.leftDeck.shutdown();
+    g4v.rightDeck.shutdown();
+
+    // Switch off leds not connected to controls
+    midi.sendShortMsg(0x90, 0x1B, 0x00);	// Left HotCue pad mode select button
+    midi.sendShortMsg(0x90, 0x1C, 0x00);	// Left Auto Loop pad mode select button
+    midi.sendShortMsg(0x90, 0x1D, 0x00);	// Left Sample pad mode select button
+    midi.sendShortMsg(0x90, 0x1E, 0x00);	// Left Loop Roll pad mode select button
+    midi.sendShortMsg(0x90, 0x26, 0x00);	// Left Deck select button
+    midi.sendShortMsg(0x91, 0x1B, 0x00);	// Right HotCue pad mode select button
+    midi.sendShortMsg(0x91, 0x1C, 0x00);	// Right Auto Loop pad mode select button
+    midi.sendShortMsg(0x91, 0x1D, 0x00);	// Right Sample pad mode select button
+    midi.sendShortMsg(0x91, 0x1E, 0x00);	// Right Loop Roll pad mode select button
+    midi.sendShortMsg(0x91, 0x26, 0x00);	// Right Deck select button
+
+    // Switch off VU metters
+    midi.sendShortMsg(0xB3, 0x14, 0x00);
+    midi.sendShortMsg(0xB3, 0x15, 0x00);
+    midi.sendShortMsg(0xB3, 0x16, 0x00);
+    midi.sendShortMsg(0xB3, 0x17, 0x00);
+    midi.sendShortMsg(0xB3, 0x18, 0x00);
+    midi.sendShortMsg(0xB3, 0x19, 0x00);
```

Comment:
since the vumeters are now components with a shutdown method, this could be removed, right?

Reply:
Done